lib/smooth/include/smooth/application/display/ST7735.h

```diff
@@ -0,0 +1,147 @@
+/*
+Smooth - A C++ framework for embedded programming on top of Espressif's ESP-IDF
+Copyright 2019 Per Malmberg (https://gitbub.com/PerMalmberg)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#pragma once
+
+#include <vector>
+#include <array>
+#include "DisplayTypes.h"
+#include "DisplayCommands.h"
+
+namespace smooth::application::display
+{
+    /************************************************************************************
+      SPECIAL NOTES                SPECIAL NOTES                          SPECIAL NOTES
+
+      The following is a collection of initialization sequences for the ST7735.  Some
+      displays will require a combination of these init sequences depending upon the tab
+      color of the protective plastic screen cover that is shipped with a new display.
+
+      Note 1.  The M5StickC uses the ST7735S which requires special int_cmds sequence.
+      The following code is an example of how to initialize the ST7735S display.
+            send_init_cmds(init_cmds_R_part1.data(), init_cmds_R_part1.size());
+            send_init_cmds(init_cmds_R_grn_tab_part2.data(), init_cmds_R_grn_tab_part2.size());
+            send_cmd(LcdCmd::INVON);  // display inversion on
+            send_init_cmds(init_cmds_R_part3.data(), init_cmds_R_part3.size());
+    ************************************************************************************/
+
+    // init commands for black tab - 17 commands
+    static const std::vector<display_init_cmd_t> init_cmds_7735B =
```

Comment:
If you make these into `static constexpr std::array<...>` you can save memory at runtime since the compiler will be able to determine which parts of the array you actually use.

Reply:
Done